lisa/tests/staging/scheduler/util_tracking.py

```diff
@@ -0,0 +1,377 @@
+        plat_info = target.plat_info
+        rtapp_profile = cls.get_rtapp_profile(plat_info)
+
+        # After a bit of experimenting, it turns out that on some platforms
+        # misprediction of the idle time (which leads to a shallow idle state,
+        # a wakeup and another idle nap) can mess up the duty cycle of the
+        # rt-app task we're running. In our case, a 50% duty cycle, 16ms period
+        # task would always be active for 8ms, but it would sometimes sleep for
+        # only 5 or 6 ms.
+        # This is fine to do this here, as we only care about the proper
+        # behaviour of the signal on running/not-running tasks.
+        with target.disable_idle_states():
+            with target.cpufreq.use_governor(**cls.cpufreq_conf):
+                cls._run_rtapp(target, res_dir, rtapp_profile, ftrace_coll)
+
+        return cls(res_dir, plat_info)
+
+    @LoadTrackingAnalysis.df_tasks_signals.used_events
+    @requires_events('sched_switch', 'sched_util_est_cfs', 'sched_util_est_task')
+    def get_task_sched_signals(self, trace, cpu, task_name):
+        """
+        Get a :class:`pandas.DataFrame` with the sched signals for the workload task
+
+        This examines scheduler utilization tracking trace events. You will need a
+        target kernel that includes the required events.
+
+        :returns: :class:`pandas.DataFrame` with a column for each signal for
+          the workload task
+        """
+        df = trace.analysis.load_tracking.df_tasks_signals()
+        df = df[df['comm'] == task_name]
+        window = self.get_task_window(trace, task_name, cpu)
+        df = select_window(df, window)
+
+        # Normalize the signal with the detected task execution start
+        df.index -= window[0]
+
+        return df
+
+    @classmethod
+    def check_from_target(cls, target):
+        super().check_from_target(target)
+
+        try:
+            target.plat_info["cpu-capacities"]
+        except KeyError as e:
+            raise CannotCreateError(str(e))
+
+    @property
+    def rtapp_profile(self):
+        return self.get_rtapp_profile(self.plat_info)
+
+class Convergence(UtilTrackingBase):
+    """
+    Basic checks for estimated utilization signals
+
+    **Expected Behaviour:**
+
+    The estimated utilization of a task is properly computed starting form its
+    `util_avg` value at the end of each activation.
+
+    Two signals composes the estimated utlization of a task:
+
+    * `util_est_enqueued` : is expected to match the max between `util_avg` and
+      `util_est_ewma` at the end of the previous activation
+
+    * `util_est_ewma` : is expected to track an Exponential Weighted Moving
+      Average of the `util_avg` signal sampled at the end of each activation.
+
+    Based on these two invariant, this class provides a set of tests to verify
+    these conditions using different methods and sampling points.
+    """
+    task_prefix = 'invariance'
```

Comment:
Unused attribute, which also kind of conflicts with the some classes of load_tracking

Reply:
Right, which also spoils the code I started from... :)